ceph/rados/monitor_workflows.py

```diff
@@ -0,0 +1,260 @@
+            Pass -> True, Fail -> false
+        """
+        cmd = f"ceph orch ps {host}"
+        out = self.rados_obj.run_ceph_command(cmd=cmd)
+        for entry in out:
+            if entry["daemon_type"] == "mon":
+                log.debug("Mon daemon present on the host")
+                return True
+        log.debug("Mon daemon not present on the host")
+        return False
+
+    def remove_mon_service(self, host) -> bool:
+        """
+        Method to remove the mon service from the given host
+
+        Args:
+            host: name of the host from where mon needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mon service exists before removing
+
+        if not self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon not present on the host {host}. Returning Pass")
+            return True
+
+        cmd = f"ceph orch daemon rm mon.{host} --force"
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be removed
+        time.sleep(10)
+
+        if self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon still present on the host {host}. Returning Fail")
+            return False
+
+        # Removing mon label from host if present
+        labels = self.get_host_labels(host=host)
+        if "mon" in labels:
+            cmd = f"ceph orch host label rm {host} mon"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Removed mon label from host : {host}")
+
+        log.debug(f"Mon successfully removed from host : {host}")
+        return True
+
+    def add_mon_service(self, **kwargs) -> bool:
+        """
+        Method to remove the mon service from the given host
+
+        Args:
+            kwargs: the accepted KW args for the method are:
+                host: name of the host from where mon needs to be removed
+                ip: IP of the mon host
+                location_type: Name of the crush bucket type if needed (optional)
+                location_name: Name of the crush location type if needed (optional)
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Checking if mon service exists before removing
+        host = kwargs.get("host")
+        if self.check_mon_exists_on_host(host=host):
+            log.info(f"Mon daemon Already present on the host {host}. Returning Pass")
+            return True
+
+        log.debug(f"Adding mon daemon on the host : {host}")
+        cmd = f"ceph mon add {host} {kwargs['ip']}"
+        if kwargs.get("location_type", False):
+            cmd = cmd + f" {kwargs['location_type']}={kwargs['location_name']}"
+        log.debug(f"Adding new mon via cmd : {cmd}")
+        self.client.exec_command(sudo=True, cmd=cmd)
+
+        # Sleeping for 10 seconds for the mon to be added
+        time.sleep(10)
+        cmd = f"ceph orch daemon add mon {host}"
```

Comment:
What's the purpose of line 188 and line 196? do we need to execute both in order to add mon?

Reply:
The initialization is mostly only for stretch cluster, as in normal RHCS, it would be enough just to run orch command to spin up a mon.